performancetest/src/main/java/org/corfudb/performancetest/LogUnitPerformanceTest.java

```diff
@@ -0,0 +1,119 @@
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.corfudb.protocols.wireprotocol.DataType;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.clients.LogUnitClient;
+import org.corfudb.util.serializer.Serializers;
+import org.junit.Test;
+import java.util.*;
+
+/**
+ * Created by Nan Zhang and Lin Dong.
+ */
+
+public class LogUnitPerformanceTest extends PerformanceTest{
+    static MetricRegistry metricRegistry;
+    Timer readTimer;
+    Timer writeTimer;
+    private static final int randomBoundary = 100;
+    private static final String METRIC_PREFIX = "corfu-perf.";
+    private static final String READ_PERCENT = "logunitReadPercent";
+    private static final String SINGLE_REQUEST = "logunitSingleRequests";
+    private static final String BATCH_REQUEST = "logunitBatchRequests";
+    private static final String BATCH_SIZE = "logunitBatchSize";
+    private static final String ENTRY_SIZE = "logEntrySize";
+
+    public LogUnitPerformanceTest() {
+        metricRegistry = CorfuRuntime.getDefaultMetrics();
+        readTimer = metricRegistry.timer(METRIC_PREFIX + "logunit-read");
+        writeTimer = metricRegistry.timer(METRIC_PREFIX + "logunit-write");
+    }
+
+    /**
+     * tests write() and read() performance of logunit client,
+     * use random to generate and send mixed requests.
+     *
+     * This test does not check correctness.
+     */
+    @Test
+    public void logunitSingleReadWrite() throws Exception {
+        setMetricsReportFlags("logunit-single");
+        Process server = runServer();
+        CorfuRuntime runtime = initRuntime();
+        int numRequests = Integer.parseInt(PROPERTIES.getProperty(SINGLE_REQUEST, "100"));
+        int readPercent = Integer.parseInt(PROPERTIES.getProperty(READ_PERCENT, "50"));
+        int entrySize = Integer.parseInt(PROPERTIES.getProperty(ENTRY_SIZE, "2048"));
+
+        LogUnitClient client = new LogUnitClient(runtime.getRouter(endPoint), 0L);
+        int address = 0;
+        byte[] payload = new byte[entrySize];
+
+        for (int i = 0; i < numRequests; i++) {
+            if (address == 0 || random.nextInt(randomBoundary) > readPercent) {
+                Timer.Context context = writeTimer.time();
+                client.write(address++, null, payload, Collections.emptyMap()).get();
+                context.stop();
+            } else {
+                int pos = random.nextInt(address);
+                Timer.Context context = readTimer.time();
+                client.read(pos).get();
+                context.stop();
+            }
+        }
+        killServer(server);
+    }
+
+    /**
+     * tests writeRange() and readAll() performance of logunit client,
+     * use random to generate and send mixed requests.
+     *
+     * This test does not check correctness.
+     */
+    @Test
+    public void logunitRangeReadWrite() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651454)